src/InstrumentationEngine/ProfilerManagerWrapper.cpp

```diff
@@ -0,0 +1,231 @@
+}
+
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::AddInstrumentationMethod(
+    _In_ BSTR bstrModulePath,
+    _In_ BSTR bstrName,
+    _In_ BSTR bstrDescription,
+    _In_ BSTR bstrModule,
+    _In_ BSTR bstrClassGuid,
+    _In_ DWORD dwPriority,
+    _Out_ IInstrumentationMethod** ppInstrumentationMethod)
+{
+    HRESULT hr = S_OK;
+    CComPtr<IInstrumentationMethod> pInstrumentationMethod;
+    IfFailRet(m_pProfilerManager->AddInstrumentationMethod(
+        bstrModulePath,
+        bstrName,
+        bstrDescription,
+        bstrModule,
+        bstrClassGuid,
+        dwPriority,
+        &pInstrumentationMethod));
+    *ppInstrumentationMethod = pInstrumentationMethod.Detach();
+    return S_OK;
+}
+
+// IProfilerManager2 Methods
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::DisableProfiling()
+{
+    return m_pProfilerManager->DisableProfiling();
+}
+
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::ApplyMetadata(_In_ IModuleInfo* pMethodInfo)
+{
+    return m_pProfilerManager->ApplyMetadata(pMethodInfo);
+}
+
+// IProfilerManager3 Methods
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::GetApiVersion(_Out_ DWORD* pApiVer)
+{
+    HRESULT hr = S_OK;
+    DWORD apiVer;
+    IfFailRet(m_pProfilerManager->GetApiVersion(&apiVer));
+    *pApiVer = apiVer;
+    return S_OK;
+}
+
+// IProfilerManager4 Methods
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::GetGlobalLoggingInstance(_Out_ IProfilerManagerLogging** ppLogging)
+{
+    HRESULT hr = S_OK;
+    CComPtr<IProfilerManagerLogging> pLogging;
+    IfFailRet(m_pProfilerManager->GetGlobalLoggingInstance(&pLogging));
+    *ppLogging = pLogging.Detach();
+    return S_OK;
+}
+
+
+// IProfilerManager5 Methods
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::IsInstrumentationMethodRegistered(_In_ REFGUID clsid, _Out_ BOOL* pfRegistered)
+{
+    HRESULT hr = S_OK;
+    BOOL fRegistered;
+    IfFailRet(m_pProfilerManager->IsInstrumentationMethodRegistered(clsid, &fRegistered));
+    *pfRegistered = fRegistered;
+    return S_OK;
+}
+
+// IProfilerManagerLogging Methods
+/*
+*/
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::LogMessage(_In_ const WCHAR* wszMessage)
+{
+    WCHAR wszPrefix[MAX_PATH];
+    wcscpy_s(wszPrefix, m_wszInstrumentationMethodGuid.c_str());
+    wcscat_s(wszPrefix, wszMessage);
+    return m_pProfilerManager->LogMessage(wszPrefix);
+}
+
+/*
+*/
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::LogError(_In_ const WCHAR* wszError)
+{
+    WCHAR wszPrefix[MAX_PATH];
+    wcscpy_s(wszPrefix, m_wszInstrumentationMethodGuid.c_str());
+    wcscat_s(wszPrefix, wszError);
+    return m_pProfilerManager->LogError(wszPrefix);
+}
+
+/*
+*/
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::LogDumpMessage(_In_ const WCHAR* wszMessage)
+{
+    WCHAR wszPrefix[MAX_PATH];
```

Comment:
You'll have to check if the tests are okay with this change. The logging system and tests treat dump logging slightly differently than error and message logging.

Reply:
You're correct, the Dumps are used to test the result of the output. However, our tests currently do not track InstrumentationMethod logging. I will add new tests to incorporate those changes.

---
In reply to: [331297278](https://github.com/microsoft/CLRInstrumentationEngine/pull/146#discussion_r331297278) [](ancestors = 331297278)

Reply:
Worked around this by making NaglerInstrumentationMethod depend on the global logging instance rather than the profiler manager wrapper. 

---
In reply to: [331297278](https://github.com/microsoft/CLRInstrumentationEngine/pull/146#discussion_r331297278) [](ancestors = 331297278)